src/jobs/jobs.controller.ts

```diff
@@ -571,52 +653,73 @@ export class JobsController {
   }
 
 
+  /**
+   * Get jobs
+   */
   @UseGuards(PoliciesGuard)
-  @CheckPolicies((ability: AppAbility) => ability.can(AuthOp.Read, JobClass))
+  @CheckPolicies((ability: AppAbility) =>
+    ability.can(AuthOp.JobsRead, JobClass),
+  )
   @Get()
+  @ApiOperation({
+    summary: "It returns a list of jobs.",
+    description: "It returns a list of jobs. The list returned can be modified by providing a filter.",
+  })
   @ApiQuery({
     name: "filter",
-    description: "Database filters to apply when retrieve all jobs",
+    description: "Filters to apply when retrieve all jobs\n" + filterDescriptionSimplified,
     required: false,
+    type: String,
+    example: filterExampleSimplified,
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+    type: [JobClass],
+    description: "Found jobs",
   })
-  async findAll(@Query("filter") filter?: string): Promise<JobClass[]> {
+  async findAll(
+    @Req() request: Request,
+    @Query("filter") filter?: string,
+  ): Promise<JobClass[] | null> {
     const parsedFilter: IFilters<
       JobDocument,
       FilterQuery<JobDocument>
     > = JSON.parse(filter ?? "{}");
+
+    if (!this.isFilterValid(Object.keys(parsedFilter))) {
```

Comment:
`isFilterValid` is a pretty limited check (keys only, top level only). Does more validation of the filter happen in the `@Query("filter")` magic?

Reply:
`isFilterValid` was created as a basic syntax check for the GET filters, because I noticed there was none. We can discuss how it can be expanded. There is no additional validation happening.